File: Day4/Tasks/catkin_ws/src/light_robot/scripts/msg_pub.py
#!/usr/bin/env python

#Msg Example presented By eng Khaled Gabr 
import rospy
from light_robot.msg import messages
import random

def talker():
    pub = rospy.Publisher('complex', messages)
    rospy.init_node('talker', anonymous=True)
    r = rospy.Rate(10) #10hz
    msg = messages()
   

    while not rospy.is_shutdown():
        msg.real=23.33 + (random.random())
        msg.imag=12 + (random.random()*2)
        rospy.loginfo("The Complex Number is %.2f + i * %.2f ",msg.real , msg.imag)
        pub.publish(msg)
        r.sleep()

if __name__ == '__main__':
    try:
        talker()
    except rospy.ROSInterruptException: pass

File: Day4/Tasks/catkin_ws/src/light_robot/src/msg_pub.cpp
/* Publisher Code for a custom message 
*/


#include "ros/ros.h"
#include "light_robot/messages.h"

#include <sstream>

/**
 * This tutorial demonstrates simple sending of messages over the ROS system.
 */
int main(int argc, char **argv)
{
  ros::init(argc, argv, "talker");

  ros::NodeHandle n;

  ros::Publisher counting_pub = n.advertise<light_robot::messages>("counting", 1000);

  ros::Rate loop_rate(10);
 
  int count = 0;
  while (ros::ok())
  {
   
    light_robot::messages msg;

    msg.real = 11+rand () ;
    msg.imag=  12+ rand () * 2 ;

    ROS_INFO("Our complex Number is  : %.2f + i * %.2f",msg.real , msg.imag);

    counting_pub.publish(msg);
  


    ros::spinOnce();

    loop_rate.sleep();
    ++count;
  }


  return 0;
}
